### DISCUSSION TITLE
Cmake Error

### DISCUSSION THREAD

**Discussion by iamstainlessironman** [NVlabs/instant-ngp#1078] [Q&A]
[ 89%] Building CUDA object CMakeFiles/ngp.dir/src/triangle_bvh.cu.o
Killed
make[2]: *** [CMakeFiles/ngp.dir/build.make:199：CMakeFiles/ngp.dir/src/common_device.cu.o] 错误 137
make[2]: *** 正在等待未完成的任务....
Killed
make[2]: *** [CMakeFiles/ngp.dir/build.make:251：CMakeFiles/ngp.dir/src/testbed.cu.o] 错误 137
/home/a/instant-ngp/src/marching_cubes.cu: In function ‘void ngp::draw_mesh_gl(const tcnn::GPUMemory<Eigen::Matrix<float, 3, 1> >&, const tcnn::GPUMemory<Eigen::Matrix<float, 3, 1> >&, const tcnn::GPUMemory<Eigen::Matrix<float, 3, 1> >&, const tcnn::GPUMemory<unsigned int>&, const Vector2i&, const Vector2f&, const Eigen::Matrix<float, 3, 4>&, const Vector2f&, int)’:
/home/a/instant-ngp/src/marching_cubes.cu:139:36: warning: ‘cudaError_t cudaGLUnregisterBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  139 |     cudaGLUnregisterBufferObject(VBO[i]);
      |                                    ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:404:46: note: declared here
  404 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnregisterBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:139:36: warning: ‘cudaError_t cudaGLUnregisterBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  139 |     cudaGLUnregisterBufferObject(VBO[i]);
      |                                    ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:404:46: note: declared here
  404 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnregisterBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:146:34: warning: ‘cudaError_t cudaGLRegisterBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  146 |    cudaGLRegisterBufferObject(VBO[i]);
      |                                  ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:327:46: note: declared here
  327 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLRegisterBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:146:34: warning: ‘cudaError_t cudaGLRegisterBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  146 |    cudaGLRegisterBufferObject(VBO[i]);
      |                                  ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:327:46: note: declared here
  327 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLRegisterBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:151:33: warning: ‘cudaError_t cudaGLUnregisterBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  151 |    cudaGLUnregisterBufferObject(els);
      |                                 ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:404:46: note: declared here
  404 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnregisterBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:151:33: warning: ‘cudaError_t cudaGLUnregisterBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  151 |    cudaGLUnregisterBufferObject(els);
      |                                 ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:404:46: note: declared here
  404 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnregisterBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:158:31: warning: ‘cudaError_t cudaGLRegisterBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  158 |   cudaGLRegisterBufferObject(els);
      |                               ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:327:46: note: declared here
  327 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLRegisterBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:158:31: warning: ‘cudaError_t cudaGLRegisterBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  158 |   cudaGLRegisterBufferObject(els);
      |                               ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:327:46: note: declared here
  327 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLRegisterBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:161:35: warning: ‘cudaError_t cudaGLMapBufferObject(void**, GLuint)’ is deprecated [-Wdeprecated-declarations]
  161 |  cudaGLMapBufferObject(&ptr, VBO[0]);
      |                                   ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:356:46: note: declared here
  356 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLMapBufferObject(void **devPtr, GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:161:35: warning: ‘cudaError_t cudaGLMapBufferObject(void**, GLuint)’ is deprecated [-Wdeprecated-declarations]
  161 |  cudaGLMapBufferObject(&ptr, VBO[0]);
      |                                   ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:356:46: note: declared here
  356 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLMapBufferObject(void **devPtr, GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:163:35: warning: ‘cudaError_t cudaGLMapBufferObject(void**, GLuint)’ is deprecated [-Wdeprecated-declarations]
  163 |  cudaGLMapBufferObject(&ptr, VBO[1]);
      |                                   ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:356:46: note: declared here
  356 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLMapBufferObject(void **devPtr, GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:163:35: warning: ‘cudaError_t cudaGLMapBufferObject(void**, GLuint)’ is deprecated [-Wdeprecated-declarations]
  163 |  cudaGLMapBufferObject(&ptr, VBO[1]);
      |                                   ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:356:46: note: declared here
  356 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLMapBufferObject(void **devPtr, GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:165:35: warning: ‘cudaError_t cudaGLMapBufferObject(void**, GLuint)’ is deprecated [-Wdeprecated-declarations]
  165 |  cudaGLMapBufferObject(&ptr, VBO[2]);
      |                                   ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:356:46: note: declared here
  356 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLMapBufferObject(void **devPtr, GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:165:35: warning: ‘cudaError_t cudaGLMapBufferObject(void**, GLuint)’ is deprecated [-Wdeprecated-declarations]
  165 |  cudaGLMapBufferObject(&ptr, VBO[2]);
      |                                   ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:356:46: note: declared here
  356 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLMapBufferObject(void **devPtr, GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:171:31: warning: ‘cudaError_t cudaGLUnmapBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  171 |  cudaGLUnmapBufferObject(VBO[2]);
      |                               ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:382:46: note: declared here
  382 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnmapBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:171:31: warning: ‘cudaError_t cudaGLUnmapBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  171 |  cudaGLUnmapBufferObject(VBO[2]);
      |                               ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:382:46: note: declared here
  382 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnmapBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:172:31: warning: ‘cudaError_t cudaGLUnmapBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  172 |  cudaGLUnmapBufferObject(VBO[1]);
      |                               ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:382:46: note: declared here
  382 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnmapBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:172:31: warning: ‘cudaError_t cudaGLUnmapBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  172 |  cudaGLUnmapBufferObject(VBO[1]);
      |                               ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:382:46: note: declared here
  382 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnmapBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:173:31: warning: ‘cudaError_t cudaGLUnmapBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  173 |  cudaGLUnmapBufferObject(VBO[0]);
      |                               ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:382:46: note: declared here
  382 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnmapBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:173:31: warning: ‘cudaError_t cudaGLUnmapBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  173 |  cudaGLUnmapBufferObject(VBO[0]);
      |                               ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:382:46: note: declared here
  382 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnmapBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:174:32: warning: ‘cudaError_t cudaGLMapBufferObject(void**, GLuint)’ is deprecated [-Wdeprecated-declarations]
  174 |  cudaGLMapBufferObject(&ptr, els);
      |                                ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:356:46: note: declared here
  356 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLMapBufferObject(void **devPtr, GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:174:32: warning: ‘cudaError_t cudaGLMapBufferObject(void**, GLuint)’ is deprecated [-Wdeprecated-declarations]
  174 |  cudaGLMapBufferObject(&ptr, els);
      |                                ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:356:46: note: declared here
  356 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLMapBufferObject(void **devPtr, GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:176:28: warning: ‘cudaError_t cudaGLUnmapBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  176 |  cudaGLUnmapBufferObject(els);
      |                            ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:382:46: note: declared here
  382 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnmapBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~
/home/a/instant-ngp/src/marching_cubes.cu:176:28: warning: ‘cudaError_t cudaGLUnmapBufferObject(GLuint)’ is deprecated [-Wdeprecated-declarations]
  176 |  cudaGLUnmapBufferObject(els);
      |                            ^
/usr/local/cuda-11.8/bin/../targets/x86_64-linux/include/cuda_gl_interop.h:382:46: note: declared here
  382 | extern __CUDA_DEPRECATED __host__ cudaError_t CUDARTAPI cudaGLUnmapBufferObject(GLuint bufObj);
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~
make[1]: *** [CMakeFiles/Makefile2:230：CMakeFiles/ngp.dir/all] 错误 2
make: *** [Makefile:103：all] 错误 2

**Comment by kameel2311** [comment]
So the process is just being killed because its using so much compute due to the -j flag , what I did is use -j4 to make 4 process max at a time (use 4 cpus max), and made it work. Can drop the whole -j but will take a long while.

Note that errors shown are only warnings and the issue we have is the process being killed. Hope this helps
